# IST 652 Lab #1
### Instructions
- Complete all 6 questions in this assignment.
- You may work with others, <b> but the work you submit must be your own </b>. You can differentiate your work by adding comments or changing the values you use to test your code. However, submitting some else's work as your own is an academic integrity violation and will be raised to academic affairs.
- It is always better to attempt a problem as partial credit may be granted.


### Submission Guide:
- Submit your answers on BlackBoard by Saturday 2019-02-02.
- The file must be either a .py or .ipynb file type.
- The name of the file should be <i> ist652_lab1_lastname.py (.ipynb) </i> .


### Grading [ 6 total points ]
For Each Question, the following credit will be awarded:
- 0.75 for printing the correct answer to the console.
- 0.15 for approaching the problem efficiently.
- 0.05 for properly documenting and commenting your code.

---
### Questions

#### ( 1 ) Write the following code and retrun <i>x</i> using the print function. What does it return and why?
- x = 11110
- x = x+1

##### [1 point]

In [ ]:
# Enter your code here, printing relevant answers to console:

x = 11110
x = x + 1

print(x)



11111


* The above code returns 11111, x is assigned a value 11110 and we are incrementing its value by 1 and printing it out.

----
#### ( 2 ) Assume that we execute the following assignment statements:
- width = 17
- height = 12.0

#### For each of the following expressions, print the value of the expression and its data type.
-  width / 2
-  width / 2.0
-  height * 2
-  height / 2.0
-  width / height


##### [1 point]

In [ ]:
# Enter your code here, printing relevant answers to console:

width =17
height  = 12.0

print("Value of width/2 : ", width/2, " ,Data Type : ",type(width/2))
print("Value of width/2.0 :",width/2.0 ,"Data Type : ",type(width/2.0))
print("Value of height*2:", height*2,"Data Type : ",type(height*2))
print("Value of height/2.0:", height/2.0 ,"Data Type : ",type(height/2.0))
print("Value of width/height:", round(width/height, 3) ,"Data Type : ",type(width/height))





Value of width/2 :  8.5  ,Data Type :  <class 'float'>
Value of width/2.0 : 8.5 Data Type :  <class 'float'>
Value of height*2: 24.0 Data Type :  <class 'float'>
Value of height/2.0: 6.0 Data Type :  <class 'float'>
Value of width/height: 1.417 Data Type :  <class 'float'>


---
#### ( 3 ) Write a sequence of statements which <u><i>prompt</i></u> the user for hours and rate per hour, printing each one, and then to computing gross pay as:
- ( hours * rate )

#### Your output lines should look something like:
- Enter Hours: 35
- Enter Rate: 2.75
- Pay: 96.25


##### [1 point]

In [3]:
# Enter your code here, printing relevant answers to console:

#typecasting input to convert input from string to float
hours = float(input("Enter Hours : "))
rate = float(input("Enter Rate per hour : "))
print("Pay : ", hours*rate)






Enter Hours : 35
Enter Rate per hour : 2.75
Pay :  96.25


 ----

#### ( 4 ) Rewrite your pay computation <i><u>(#3)</u></i> to give the employee 1.5 times the hourly rate for hours worked above 40 hours.
#### For example:
- Hours: 45
- Rate: 10
- Pay: 475.0


##### [1 point]

In [ ]:
# Enter your code here, printing relevant answers to console:

#typecasting input to convert input from string to float
hours = float(input("Hours: ")) 
rate = float(input("Rate: "))

# Conditions for pay calculation
if(hours > 40):
    overtime_hours = hours - 40
    overtime_rate = rate * 1.5
    overtime_pay = overtime_hours * overtime_rate
    regular_pay = 40 * rate
    total_pay = overtime_pay + regular_pay
else:
    total_pay = hours * rate
    
print("Pay:",total_pay)







Hours: 45
Rate: 10
Pay: 475.0


Explanation: 

* The code here takes hours and rate as inputs. The control goes into 
the if block first if the number of hours entered is > 40. In the if block, first the number of extra hours worked(hours - 40) and the extra rate ie., 1.5 times the actual rate , are calculated. Next, the pay for 40 hours worked is calculated. The sum of pay for extra hours and 40 hours is summed to get the final pay.
* If the number of hours worked is less 40 then the normal pay is calculated


---

#### ( 5 ) Suppose that there is a list of strings defined, called samples. Define the list so that some strings have only 1 or 2 characters and some strings have more.  Write a loop that prints out all the strings whose length is greater than 2. 

Nake up a list then write your loop to use the list samples, for example:

   <i> samples = [‘at’, ‘bat’, ‘c’, . . . .  ]  </i>

Submit your list, your code and an example run.


##### [1 point]

In [ ]:
# Enter your code here, printing relevant answers to console:

samples = ['I', 'am', 'a', 'data', 'scientist', 'in', 'the', 'making', 'with', 'passion', 'for', 'ML']

#Loop to check for lengths of each list element, printing word only out if length > 2

for x in samples:
  if len(x) > 2 :
    print(x)
  else: continue   #optional line
 





data
scientist
the
making
with
passion
for


#### ( 6 ) Again suppose that there is a list of strings defined, called samples.  Define the list so that some strings have only 1 or 2 characters and some strings have <u>more than 5.</u>  Write a loop that prints out all the strings whose length is greater than 2 and whose length is less than 5. 

Make up a list then write your loop to use the list samples. For example:

<i>samples = [‘at’, ‘book’, ‘c’, ‘dog’, ‘elephant’, . . .  ]</i>

Submit your list, your code and an example run.


##### [1 point]

In [ ]:
# Enter your code here, printing relevant answers to console:

samples = ['I', 'am', 'a', 'data', 'scientist', 'in', 'the', 'making', 'with', 'passion', 'for', 'ML']

#Loop to check for lengths of each list element, printing word only out if length > 2 and <5
for x in samples:
  if len(x) >2 and len(x)<5:
    print(x)
  else: continue








data
the
with
for
